packages/lib-sourcify/src/Verification/Verification.ts

```diff
@@ -0,0 +1,500 @@
+      );
+    }
+
+    // We need to manually generate the auxdata positions because they are not automatically produced during compilation
+    // Read more: https://docs.sourcify.dev/blog/finding-auxdatas-in-bytecode/
+    await this.compilation.generateCborAuxdataPositions();
+
+    // Try to match onchain runtime bytecode with compiled runtime bytecode
+    try {
+      logDebug('Matching with deployed bytecode', {
+        chain: this.sourcifyChain.chainId,
+        address: this.address,
+      });
+      await this.matchWithRuntimeBytecode();
+    } catch (e: any) {
+      logWarn('Error matching with runtime bytecode', {
+        chain: this.sourcifyChain.chainId,
+        address: this.address,
+        error: e.message,
+      });
+    }
+
+    const solidityBugType = await this.handleSolidityBugCases(
+      forceEmscripten,
+      compiledRuntimeBytecode,
+    );
+    if (solidityBugType === SolidityBugType.ECMASCRIPT_BUG) {
+      // TODO: change name of ECMASCRIPT_BUG
+      return await this.verify({ forceEmscripten: true });
+    }
+
+    // Try to match onchain creation bytecode with compiled creation bytecode
+    if (this.creatorTxHash) {
+      try {
+        logDebug('Matching with creation tx', {
+          chain: this.sourcifyChain.chainId,
+          address: this.address,
+          creatorTxHash: this.creatorTxHash,
+        });
+
+        // Get creation transaction data
+        const creatorTx = await this.sourcifyChain.getTx(this.creatorTxHash);
+        this.blockNumber = creatorTx.blockNumber || undefined;
+        this.deployer = creatorTx.from;
+
+        const { creationBytecode, txReceipt } =
+          await this.sourcifyChain.getContractCreationBytecodeAndReceipt(
+            this.address,
+            this.creatorTxHash,
+            creatorTx,
+          );
+        this._onchainCreationBytecode = creationBytecode;
+        this.txIndex = txReceipt.index;
+
+        await this.matchWithCreationTx();
+      } catch (e: any) {
+        logWarn('Error matching with creation tx', {
+          chain: this.sourcifyChain.chainId,
+          address: this.address,
+          creatorTxHash: this.creatorTxHash,
+          error: e.message,
+        });
+      }
+    }
+
+    if (this.creationMatch !== null || this.runtimeMatch !== null) {
+      logInfo('Verified contract', {
+        address: this.address,
+        chainId: this.sourcifyChain.chainId,
+        runtimeMatch: this.runtimeMatch,
+        creationMatch: this.creationMatch,
+      });
+      return;
+    }
+
+    throw new VerificationError(
+      "The deployed and recompiled bytecode don't match.",
+      'NO_MATCH',
+    );
+  }
+
+  async handleSolidityBugCases(
+    forceEmscripten: boolean,
+    compiledRuntimeBytecode: string,
+  ): Promise<SolidityBugType> {
+    // Handle Solidity specific verification bug cases
+    if (
+      this.compilation instanceof SolidityCompilation &&
+      this.runtimeMatch === null
+    ) {
```

Comment:
IMO this should be outside the function. It was weird to read handleSolidityBugCases in the verify function wihtout these checks

Reply:
Sorry but saw this is resolved and not moved outside?

Reply:
It is moved outside

https://github.com/ethereum/sourcify/blob/ed4fb932ea9c74ddc87f8718b7f6021a18b79594/packages/lib-sourcify/src/Verification/Verification.ts#L115-L126

Reply:
Ok sorry my bad. I was looking at the old code I guess. Can be closed